### PR TITLE
Create multiple built bundles using different .env files

### PR DESCRIPTION
**Pull request by dr3**
Resolves https://github.com/bbc/simorgh-infrastructure/issues/189

**Overall change:** Creates committed .env files for each running environment and uses environment variables `APP_ENV` to dictate where a code is bundled to run, and what bundles to use.

**Code changes:**

- Renames and creates .env files for each environment
- Passes `APP_ENV` to the `npm run build` command
- Passes `APP_ENV` to the `npm run dev` command
- On run time uses the `.env` file by default unless `process.env.APP_ENV` is set. On the ec2 `.env.test` or `.env.live` replaces `.env` file which is used to run.

To run locally 
`npm run build:local && npm start` -> localhost
`npm run dev` -> localhost
`npm run build && APP_ENV=test npm start` -> test.bbc.files etc
`npm run build && APP_ENV=live npm start` -> bbc.files etc

This PR pairs with https://github.com/bbc/simorgh-infrastructure/pull/226

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "amp:validate": "wait-on http://localhost:7080/news/articles/c9rpqy7pmypo.amp && amphtml-validator http://localhost:7080/news/articles/c9rpqy7pmypo.amp",
     "audit:ci": "audit-ci -low",
     "bbcA11y": "bbc-a11y",
-    "build": "rm -rf build && NODE_ENV=production webpack",
+    "build": "rm -rf build && NODE_ENV=production APP_ENV=test webpack && NODE_ENV=production APP_ENV=live webpack",
```
Comment (pjlee11) [inline package.json]:
```suggestion
    "build": "export  NODE_ENV=production && rm -rf build && APP_ENV=test webpack && APP_ENV=live webpack",
```

Can't decide if this is more explicit or more confusing, but it's a suggestion either way...

File: src/server/env/index.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  it('calls dotenv with correct path when APP_ENV is "local"', async () => {
+    process.env.APP_ENV = 'local';
+
+    const result = getEnv();
+
+    expect(dotenv.config).toHaveBeenCalledWith({ path: '.env' });
+    expect(result).toEqual({ foo: 'bar' });
+  });
+
+  it('calls dotenv with correct path when APP_ENV is "local"', async () => {
```
Comment (pjlee11) [inline src/server/env/index.test.js]:
```suggestion
  it('calls dotenv with correct path when APP_ENV is "test"', async () => {
```

File: webpack.config.js [hunk 3]
```diff
@@ -2,20 +2,17 @@
...
-
-if (result.error) {
-  throw result.error;
-}
+// Load in environment variables configured in `.env` files.
+const DOT_ENV_CONFIG = getEnv();
 
 // `shell` parameter populated via CLI, e.g. --env.platform=web
 module.exports = (shell = {}) => {
+  const APP_ENV = process.env.APP_ENV || 'live';
```
Comment (pjlee11) [inline webpack.config.js:15]: By having this default to live doesn't that make the logic to fall back to `.env` if `process.env.APP_ENV` is not set obsolete? 

EG: https://github.com/bbc/simorgh/pull/1204/files#diff-3891f3f26878f8e17bd04f1a222621bbR7

**Review by pjlee11 — CHANGES_REQUESTED** [review]
Great start, just a few comments. Also think CodeClimate is right and the tests for `src/server/env/index.test.js` could be DRY'd

File: webpack.config.js (hunk 3 above)
Comment (dr3) [reply webpack.config.js:15]: This is just a safeguard incase https://github.com/bbc/simorgh/blob/WebpackEnviromentMultiBuild/.env#L5 doesnt exist for some reason

**Review by pjlee11 — DISMISSED** [review]

**Review by sareh — APPROVED** [review]
Brilliant work, Drew! 🎉

**Review by pjlee11 — APPROVED** [review]

**Comment by ChrisBAshton** [comment]
Code reviewed ✅ 

Regression tested: `npm run dev` still renders article as expected, Storybook continues to work. ✅

Feature tested: `npm run build` creates `assets.live.json` and `assets.test.json`, which reference news.files and news.test.files respectively. Different hashed bundles exist under `public/static/js`, and the live and test assets manifest _does_ reference different bundles between environments. Searching the 'test' bundle for `test.bbc`, I find entries - and when I search `test.bbc` on the 'live' bundle, I can't find any matches, as expected. ✅

This PR assumes that the Test and Live bundles are both deployed to the Test bucket, and both deployed to the Live bucket. [We'll tidy that up later](https://github.com/bbc/simorgh-infrastructure/pull/231#pullrequestreview-197100725).

**Pull request merged**